storage/mdadm/include/include.sh

```diff
@@ -0,0 +1,619 @@
+#!/bin/bash
+
+# Copyright (c) 2016 Red Hat, Inc. All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Author: Chnaghui Zhong <czhong@redhat.com>
+
+rpm -q mdadm || yum install -y mdadm
+JOURNAL_SUPPORT=0
+info=`mdadm --create --help | grep -o "write-journal"`
+if [ "$info" = "write-journal" ]; then
+        JOURNAL_SUPPORT=1
+fi
+
+#Install fio from upstream
+function install_fio() {
+
+	git_url=git://git.kernel.org/pub/scm/linux/kernel/git/axboe/fio.git
+
+	yum install libaio-devel zlib-devel -y
+	git clone $git_url
+	echo "INFO: Installing Fio"
+	cd fio  &&./configure && make && make install
+	which fio
+	if [ $? -ne 0 ]; then
+		echo "FAIL: Fio not succesffully installed"
+		exit 1
```

Comment:
Is this a test prerequisite or is the test also testing the ability to install fio? If it's just a prerequisite we don't want to report this failure to developers and the test should abort instead (see the top level readme for details). I don't see the function being used in this PR though.

Reply:
Look like this would qualify as a setup/infra failure, we need to abort the recipe in these cases, see https://github.com/CKI-project/tests-beaker/blob/master/acpi/acpitable/runtest.sh#L34